### PR TITLE
Upgrade to Rust SGX SDK 2.0.0

### PR DESCRIPTION
**Pull request by henrysun007**
## Description

Upgrade Rust toolchain and SGX software.

Fixes # (issue)

https://github.com/apache/incubator-teaclave/issues/664

## Type of change (select or add applied and delete the others)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage
- [ ] Code cleanup or just sync with upstream third-party crates

## How has this been tested?
CI and local HW tests.

## Checklist

- [X] Fork the repo and create your branch from `master`.
- [ ] If you've added code that should be tested, add tests.
- [ ] If you've changed APIs, update the documentation.
- [X] Ensure the tests pass (see CI results).
- [X] Make sure your code lints/format.

### CODE REVIEW

**Comment by volcano0dr** [comment]
Rust SGX SDK 2.0.0 has implemented [sgx_protected_fs](https://github.com/apache/incubator-teaclave-sgx-sdk/tree/v2.0.0-preview/sgx_protected_fs) with Rust, and can support the use inside and outside the Enclave. I think you can replace [protected_fs_rs](https://github.com/apache/incubator-teaclave/tree/master/common/protected_fs_rs) with sgx_protected_fs in 2.0.0.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,14 +45,15 @@ jobs:
           cd sdk/swift/TeaclaveClientSDK && xcodebuild -scheme TeaclaveClientSDK
   format:
     runs-on: ubuntu-20.04
-    container: teaclave/teaclave-build-ubuntu-1804-sgx-2.14:0.1.5
+    container: ghcr.io/henrysun007/teaclave-build-ubuntu-2004-sgx-2.17.1:upgrade
```
Comment (mssun) [inline .github/workflows/ci.yml:48]: I'll add you to the `teaclave` org in DockerHub. Let's use this org to host the image.

Comment (henrysun007) [reply .github/workflows/ci.yml:48]: I was added to `teaclave` organization in docker Hub, but seem to have no permission to create new repos.

Comment (mssun) [reply .github/workflows/ci.yml:49]: I have invited you as the admin. Can you try again? Thanks!

Comment (mssun) [reply .github/workflows/ci.yml:48]: I have added you as the owner. Please try again.

Comment (henrysun007) [reply .github/workflows/ci.yml]: The pushing succeeded this time. Thanks, mingshen.

**Comment by henrysun007** [comment]
@mssun I cannot open PR to https://github.com/apache/incubator-teaclave-crates as it is empty. So I pushed the crates to https://github.com/mesatee/crates. The code is tested. You can upload the code directly to `incubator-teaclave-crates`. Let me know after the upload.

**Comment by mssun** [comment]
> @mssun I cannot open PR to https://github.com/apache/incubator-teaclave-crates as it is empty. So I pushed the crates to https://github.com/mesatee/crates. The code is tested. You can upload the code directly to `incubator-teaclave-crates`. Let me know after the upload.

I created a README.md file. You can now submit PR.

**Comment by henrysun007** [comment]
Cool. The PR is submitted. @mssun

**Review by mssun — APPROVED** [review]
LGTM.

**Pull request merged**